File: app/src/main/java/com/chiului/android_mvvm_architecture/adapter/MainAdapter.java
package com.chiului.android_mvvm_architecture.adapter;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.chiului.android_mvvm_architecture.ui.HomeFragment;
import com.chiului.android_mvvm_architecture.ui.ListSampleFragment;

/**
 * 主页的 ViewPager 2 适配器$
 *
 * @author 神经大条蕾弟
 * @date 2020/08/05 12:24
 */
public class MainAdapter extends FragmentStateAdapter {

    public MainAdapter(@NonNull Fragment fragment) {
        super(fragment);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        // Return a NEW fragment instance in createFragment(int)
        Fragment fragment;
        switch (position){
            case 3:
                fragment = ListSampleFragment.newInstance();
                break;
            case 2:
            case 1:
            case 0:
            default:
                fragment = new HomeFragment();
                Bundle args = new Bundle();
                // Our object is just an integer :-P
                args.putString(HomeFragment.ARG_OBJECT, String.valueOf(position + 1));
                fragment.setArguments(args);
                break;
        }
        return fragment;

    }

    @Override
    public int getItemCount() {
        return 4;
    }
}

File: app/src/main/java/com/chiului/android_mvvm_architecture/adapter/MainAdapter.kt
package com.chiului.android_mvvm_architecture.adapter

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.chiului.android_mvvm_architecture.ui.HomeFragment
import com.chiului.android_mvvm_architecture.ui.ListSampleFragment
import com.chiului.android_mvvm_architecture.ui.MineFragment

/**
 * 主页 Adapter$
 * @author    神经大条蕾弟
 * @date      2021/02/05 14:42
 */
class MainAdapter constructor(fragment: Fragment) : FragmentStateAdapter(fragment) {

    override fun getItemCount(): Int {
        return 4
    }

    override fun createFragment(position: Int): Fragment {
        // Return a NEW fragment instance in createFragment(int)
        val args = Bundle()
        // Our object is just an integer :-P
        args.putString(HomeFragment.ARG_OBJECT, (position + 1).toString())

        return when (position) {
            2 -> ListSampleFragment.getInstance()
            3 -> MineFragment.getInstance()
            else -> {
                val homeFragment = HomeFragment()
                homeFragment.arguments = args
                return homeFragment
            }
        }
    }
}